pkg/config/nodetreemodel/leaf_node.go

```diff
@@ -54,56 +54,12 @@ func (n *leafNodeImpl) GetChild(key string) (Node, error) {
 	return nil, fmt.Errorf("can't GetChild(%s) of a leaf node", key)
 }
 
-// GetAny returns the scalar as an interface
-func (n *leafNodeImpl) GetAny() (interface{}, error) {
-	return n.val, nil
-}
-
-// GetBool returns the scalar as a bool, or an error otherwise
-func (n *leafNodeImpl) GetBool() (bool, error) {
-	return toBool(n.val)
-}
-
-// GetInt returns the scalar as a int, or an error otherwise
-func (n *leafNodeImpl) GetInt() (int, error) {
-	return toInt(n.val)
-}
-
-// GetFloat returns the scalar as a float64, or an error otherwise
-func (n *leafNodeImpl) GetFloat() (float64, error) {
-	return toFloat(n.val)
-}
-
-// GetString returns the scalar as a string, or an error otherwise
-func (n *leafNodeImpl) GetString() (string, error) {
-	return toString(n.val)
-}
-
-// GetTime returns the scalar as a time, or an error otherwise, not implemented
-func (n *leafNodeImpl) GetTime() (time.Time, error) {
-	return time.Time{}, fmt.Errorf("not implemented")
-}
-
-// GetDuration returns the scalar as a duration, or an error otherwise, not implemented
-func (n *leafNodeImpl) GetDuration() (time.Duration, error) {
-	return time.Duration(0), fmt.Errorf("not implemented")
-}
-
-// SetWithSource assigns a value in the config, for the given source
-func (n *leafNodeImpl) SetWithSource(newValue interface{}, source model.Source) error {
-	// TODO: enforce type-checking, return an error if type changes
-	n.val = newValue
-	n.source = source
-	// TODO: Record previous value and source
-	return nil
+// Get returns the scalar as an interface
```

Comment:
Comment is out of date: (1) not necessarily a scalar, (2) no need to say "interface" when there's no longer other methods that use more specific types.

How about:
```suggestion
// Get returns the setting value stored by the leaf
```